core/src/main/java/com/linecorp/armeria/common/stream/StreamMessage.java

```diff
@@ -676,4 +677,24 @@ default EventExecutor defaultSubscriberExecutor() {
         requireNonNull(function, "function");
         return new RecoverableStreamMessage<>(this, function, /* allowResuming */ true);
     }
+
+    /**
+     *  Provide general way to use {@link StreamMessages} by applying the specified {@link Function}.
+     *
+     *  <p>Example:<pre>{@code
+     *  ByteBuf[] bufs = new ByteBuf[10];
+     *  for(int i=0; i<10; i++){
+     *             bufs[i] = Unpooled.wrappedBuffer(Integer.toString(i).getBytes());
+     *   }
+     *  StreamMessage<ByteBuf> streamMessage = StreamMessage.of(bufs);
+     *  streamMessage.writeTo( x -> HttpData.wrap())
```

Comment:
*  ByteBuf[] bufs = new ByteBuf[10]; // -> List<ByteBuf> bufs = new ArrayList<>();
     *  for(int i=0; i<10; i++){ // -> for (int i = 0; i < 10; i++)
     *       bufs.add(Unpooled.wrappedBuffer(Integer.toString(i).getBytes())); // indent, 4 space for tab
     *   }
     *  StreamMessage<ByteBuf> streamMessage = StreamMessage.of(bufs);
     *  streamMessage.writeTo( x -> HttpData.wrap()) // (x -> HttpData.wrap(x))
     * + assert~ // where is assert?

Reply:
> ```
>  *  ByteBuf[] bufs = new ByteBuf[10]; // -> List<ByteBuf> bufs = new ArrayList<>();
>  *  for(int i=0; i<10; i++){ // -> for (int i = 0; i < 10; i++)
>  *       bufs.add(Unpooled.wrappedBuffer(Integer.toString(i).getBytes())); // indent, 4 space for tab
>  *   }
>  *  StreamMessage<ByteBuf> streamMessage = StreamMessage.of(bufs);
>  *  streamMessage.writeTo( x -> HttpData.wrap()) // (x -> HttpData.wrap(x))
>  * + assert~ // where is assert?
> ```

test